### DISCUSSION TITLE
pg_cron scheduled jobs never execute in older project, but do execute in new project

### DISCUSSION THREAD

**Discussion by Musilix** [supabase/supabase#22683] [Questions] [answered]
### My Issue
For some reason it seems like the jobs I'm scheduling with the pg_cron extension are never being touched.

It seemed my issue was one-to-one with a previous issue brought up [here](https://github.com/orgs/supabase/discussions/1628), but even after following the directions of the accepted answer, none of the jobs I schedule end up doing anything. My _cron.job_run_details_ table is empty regardless of how long I sit around and wait!

### Extra Details:
* I am running all these queries (listed out in the linked dicussion) under the _**postgres**_ role.
* I created my project back around December, 2023 I believe.

I will note that when I go:
* Create a new org
* Create a new project under that org
* Enable the pg_cron extension in the project (through the UI dashboard) and run the following commands below, **it works**:
```
CREATE TABLE test_table (
  id bigint GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
  inserted_at timestamp with time zone DEFAULT timezone('utc'::text, now()) NOT NULL,
  updated_at timestamp with time zone DEFAULT timezone('utc'::text, now()) NOT NULL,
  name text
);

select
  cron.schedule(
    'whack-every-minute', 
    '* * * * *', 
    $$
    insert into test_table(name) values ('kek');
    $$
  );
```

Does anyone have an idea on what the hecks goin on? 

Something I noticed that seemed a little strange and may be of use is that when I enable pg_cron in my main project through the UI dashboard, it gives me a different modal than when I'm enabling it in the dumby project I set up.

### Modal in Main Project
![chrome_9QIF5l2Bx0](https://github.com/supabase/supabase/assets/20892398/f24ddb35-f086-421b-afc8-966dd4ab4e15) ![chrome_F926cfNnGf](https://github.com/supabase/supabase/assets/20892398/559d205f-60b2-47d8-abca-1ac18ea3b153)

### Modal in Dumby Project
![chrome_2yGdeDLGyy](https://github.com/supabase/supabase/assets/20892398/9bee6b05-8e9d-4746-9bc8-7ae1701e4583)

Regardless, even if I manually type out the 'pg_catalog' schema or 'cron' schema in the modal shown in my main project, the schedule cron jobs still produce nothing.

---
Could an extension in my main project be causing a conflict with pg_cron? 
Something to do with the age of my main project? 
Something missing in my main project? 

I have no clue, but the fact that it worked on the dumby project I made is strange.

**Comment by GaryAustin1** [comment]
There were several users with pg_cron issues with instances from a certain point in time (if I recall correctly).

Here was one thread: https://github.com/supabase/supabase/issues/19322

You should also put a support request in.

**Reply by Musilix** [reply, marked as answer]
Thanks for the quick reply!

I looked deeper in it and realized my project was using an older version of postgres. I updated to postgres v15.1.1.41 and it magically resolved the issue.

I'm unsure why, but if anyone else is facing the same issue, even after going through all the steps listed in thread #1628, then you may need to update the version of postgres you're using.
